sass-style-guide.md

```diff
@@ -59,87 +42,111 @@ You should order your rules from generic to specific. Example:
 
 // Abstract Component Imports
 //---------------------------
-@import abstract-components/section-box
+@import abstract-components/boundary-box
+@import abstract-components/gradient-overlay
 
 // Generic Component Imports
 //--------------------------
 @import generic-components/button
-@import generic-components/skip-links
+@import generic-components/data-table
 
 // Specific Component Imports
 //---------------------------
-@import specific-components/page/page-footer
-@import specific-components/page/page-header
+@import specific-components/company-search
+@import specific-components/page-header
+```
+
+## Style-Definitionen
+
+Wann wird wo was definiert? Wir gehen inhaltsgetrieben vor, d.h. wir schreiben unseren Sass-Code in der Reihenfolge, wie die Elemente auf der gerenderten Seite erscheinen werden:
+
+1. **von außen nach innen**: ZASAF folgend beginnt man in der Regel mit dem Page-Wrapper-Element.
+2. **von oben nach unten**: Danach unterteilt man sich die Seite in (horizontale) Sektionen und beginnt mit der ersten (page-header -> page-main -> page-footer).
+3. **von links nach rechts**: In einer Sektion fängt man ganz links mit dem ersten Element an (Page-Header: Logo links -> Navigation in der Mitte -> Suche rechts).
+
+### Notationsreihenfolge innerhalb eines Elements
+
+Verschachtelungen innerhalb eines Elements beginnen mit Pseudoklassen, gefolgt von Pseudoelementen, gefolgt von Modifiern. Auf Kindelemente, sowohl direkte als auch tiefer im DOM verschachtelte, sollte weitestgehend verzichtet werden. Hierfür wäre es besser einen neuen Selektorblock zu erstellen (siehe Coding Konventionen unten).
+
+``` sass
+.main-navigation--item
+  position: relative
+  &:hover
+    border-color: $color-border-loud
+  &::before
+    content: '*'
+  &.highlight
+    bacground-color: $color-background-loud
+
+.main-navigation--item
+  display: block
+  padding: 5px 0
+  .highlight > &
```

Comment:
Die Grundregel
> steht nie alleine, sondern ist nur in Verbindung mit der Komponente eindeutig

habe ich eher so verstanden, dass ich folgendes _nicht_ machen darf:

``` sass
.highlighted
  background-color: yellow
```

In meinem Beispiel steht `.highlighted` demnach auch nicht alleine, sondern bezieht das Nachfahrenelement mit ein. Apropos, Ich nehme an, in deinem obigen Sass-Beispiel meinst du eigentlich:

``` sass
.tag-cloud--link
  .highlighted &
    background-color: yellow
```

In dem Fall versteh ich deinen Einwand, das wäre Mist. Wenn du dir mein Beispiel oben aber nochmal näher anschaust, dann habe ich ein `>` benutzt, was das Problem löst.

Ich finde übrigens auch, dass

``` sass
.tag-cloud--link
  .tag-cloud--item.highlighted &
    background-color: yellow
```

schwerer zu lesen ist. Der Modifier, auf den es mir letztendlich ankommt, geht unter.

Und ja, an dieser Stelle wünscht man sich BEM. Hier würde der hervorgehobene Link eindeutig mittels `tag-cloud__link--highlighted` selektiert werden können. Ach, wie war das nochmal mit _CSS Modules_? ;)